src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -0,0 +1,1850 @@
+    snapshotStreamArgs.maxResolution = commandData.maxResolution;
+    snapshotStreamArgs.quality       = commandData.quality;
+
+    // Call the delegate
+    Status status = mDelegate.SnapshotStreamAllocate(snapshotStreamArgs, snapshotStreamID);
+
+    if (status != Status::Success)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    response.snapshotStreamID = snapshotStreamID;
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+void CameraAVStreamMgmtServer::HandleSnapshotStreamDeallocate(HandlerContext & ctx,
+                                                              const Commands::SnapshotStreamDeallocate::DecodableType & commandData)
+{
+
+    auto & snapshotStreamID = commandData.snapshotStreamID;
+
+    // Call the delegate
+    Status status = mDelegate.SnapshotStreamDeallocate(snapshotStreamID);
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+}
+
+void CameraAVStreamMgmtServer::HandleSetStreamPriorities(HandlerContext & ctx,
+                                                         const Commands::SetStreamPriorities::DecodableType & commandData)
+{
+
+    auto & streamPriorities = commandData.streamPriorities;
+    std::vector<StreamUsageEnum> rankedStreamPriorities;
+    auto iter = streamPriorities.begin();
+
+    while (iter.Next())
+    {
+        auto & streamUsage = iter.GetValue();
+        if (streamUsage == StreamUsageEnum::kUnknownEnumValue)
+        {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+            return;
+        }
+        rankedStreamPriorities.push_back(streamUsage);
+    }
+
+    CHIP_ERROR err = SetRankedVideoStreamPriorities(rankedStreamPriorities);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+        return;
+    }
+
+    mDelegate.OnRankedStreamPrioritiesChanged();
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Success);
+}
+
+void CameraAVStreamMgmtServer::HandleCaptureSnapshot(HandlerContext & ctx,
+                                                     const Commands::CaptureSnapshot::DecodableType & commandData)
+{
+
+    Commands::CaptureSnapshotResponse::Type response;
+    auto & snapshotStreamID    = commandData.snapshotStreamID;
+    auto & requestedResolution = commandData.requestedResolution;
+    ImageSnapshot image;
+
+    // Call the delegate
+    Status status = mDelegate.CaptureSnapshot(snapshotStreamID, requestedResolution, image);
+
+    if (status != Status::Success)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    if (image.data.size() > kMaxSnapshotImageSize)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ResourceExhausted);
+        return;
+    }
+
+    // copy the image data into the response
+    memcpy(&response.data, image.data.data(), image.data.size());
```

Comment:
You just smashed the stack.  This is a critical security bug right here.  Please let me know if you can't figure out why and how to do this part right.

Reply:
Presumed the response.data to be a byte array. Changed to a ByteSpan.